server/auth/types/authentication_type.ts

```diff
@@ -153,6 +151,11 @@ export abstract class AuthenticationType implements IAuthenticationType {
           return toolkit.notHandled();
         }
 
+        // allow optional authentication
+        if (this.authOptional(request)) {
+          return toolkit.authenticated();
+        }
```

Comment:
Can you also explain how is this different/why is this not already handled by the existing code here? https://github.com/opensearch-project/security-dashboards-plugin/blob/06b2e962e29df68916f53ed8d23fcaee9ee0e39d/server/auth/types/authentication_type.ts#L102

Reply:
Oh I just re-read the code, seems like you need/want the code getting the auth to execute prior to early return.

Reply:
Yes, if the user has not logged in, the code needs to keep going down, otherwise an error `No Authorization` will be reported, and no one can log in.